File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/MotorTesting.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.hardware.DcMotorSimple;
import com.qualcomm.robotcore.hardware.Servo;
import com.qualcomm.robotcore.hardware.DcMotor;

// Motor and arm tester

@TeleOp(name="Motor Tester", group="Dev")
public class MotorTesting extends LinearOpMode {

    // Declare members
    private FourWheelRobot robot;

    private DcMotor armLeft;
    private DcMotor armRight;

    @Override
    public void runOpMode() {
        robot = new FourWheelRobot(hardwareMap);
        robot.reset();

        armLeft = RobotUtil.getDcMotor(
            hardwareMap,
            "armLeft",
            DcMotorSimple.Direction.REVERSE,
            DcMotor.ZeroPowerBehavior.FLOAT,
            DcMotor.RunMode.RUN_USING_ENCODER
        );
        armRight = RobotUtil.getDcMotor(
            hardwareMap,
            "armRight",
            DcMotorSimple.Direction.FORWARD,
            DcMotor.ZeroPowerBehavior.FLOAT,
            DcMotor.RunMode.RUN_USING_ENCODER
        );

        telemetry.addData("Status", "Initialized");
        telemetry.update();

        // Wait for the game to start (driver presses PLAY)
        waitForStart();

        while (opModeIsActive()) {
            telemetry.addData("arm left", armLeft.getCurrentPosition());
            telemetry.addData("arm right", armRight.getCurrentPosition());
            telemetry.addData("left front wheel", robot.leftFront.getCurrentPosition());
            telemetry.addData("left trigger", gamepad1.left_trigger);
            telemetry.update();
        }
    }

    private double armPosition;
    private final double maxArmPos = 100.0;
    private final double minArmPos = 0.0;

    private void initArm() {
        setArmPosition(minArmPos);
        armLeft.setMode(DcMotor.RunMode.RUN_TO_POSITION);
        armRight.setMode(DcMotor.RunMode.RUN_TO_POSITION);
        armLeft.setPower(0.5);
        armRight.setPower(0.5);
    }
    private void setArmPosition(double ticks) {
        // Clamp arm position between min and max.
        // ticks = Math.max(minArmPos, Math.min(ticks, maxArmPos));

        int output = (int) Math.round(ticks);
        armLeft.setTargetPosition(output);
        armRight.setTargetPosition(output);
        armPosition = ticks;
    }
    private void shiftArmPosition(double dTicks) {
        setArmPosition(armPosition + dTicks);
    }



}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/MotorTesting.kt
package org.firstinspires.ftc.teamcode

import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode
import com.qualcomm.robotcore.hardware.DcMotor
import com.qualcomm.robotcore.hardware.Servo
import com.qualcomm.robotcore.hardware.DcMotorSimple

// Motor and arm tester
@TeleOp(name = "Motor Tester", group = "Dev")
class MotorTesting : LinearOpMode() {
    // Declare members
    private var robot: MadMachinesRobot by LateInitConstProperty()

    override fun runOpMode() {
        robot = FourWheelRobot(hardwareMap)
        robot.reset()

        telemetry.addData("Status", "Initialized")
        telemetry.update()

        // Wait for the game to start (driver presses PLAY)
        waitForStart()
        while (opModeIsActive()) {
            when {
                gamepad1.a -> {
                    robot.clawLeft.position = 0.0
                    robot.clawRight.position = 0.0
                }
                gamepad1.b -> {
                    robot.clawLeft.position = 1.0
                    robot.clawRight.position = 1.0
                }
            }
            val thingsToPrint: Array<String, Any?> = arrayOf(
                "arm left" to robot.armLeft.currentPosition,
                "arm right" to robot.armRight.currentPosition,
                "left front wheel" to robot.leftFront.currentPosition,
                "left trigger" to gamepad1.left_trigger,
                "left claw" to robot.clawLeft.position,
                "right claw" to robot.clawRight.position,
            )
            for ((caption, content) in thingsToPrint)
                telemetry.addData(caption, content)
            telemetry.update()
        }
    }
}
